.github/workflows/manual-upgrade-testnet-l2.yml

```diff
@@ -0,0 +1,149 @@
+        run: |
+          echo "L2_ENCLAVE_DOCKER_BUILD_TAG=testnetobscuronet.azurecr.io/obscuronet/enclave:latest" >> $GITHUB_ENV
+          echo "L2_HOST_DOCKER_BUILD_TAG=testnetobscuronet.azurecr.io/obscuronet/host:latest" >> $GITHUB_ENV
+          echo "L2_CONTRACTDEPLOYER_DOCKER_BUILD_TAG=testnetobscuronet.azurecr.io/obscuronet/contractdeployer:latest" >> $GITHUB_ENV
+          echo "L2_HARDHATDEPLOYER_DOCKER_BUILD_TAG=testnetobscuronet.azurecr.io/obscuronet/hardhatdeployer:latest" >> $GITHUB_ENV
+          echo "RESOURCE_TAG_NAME=testnetlatest" >> $GITHUB_ENV
+          echo "RESOURCE_STARTING_NAME=T" >> $GITHUB_ENV
+          echo "RESOURCE_TESTNET_NAME=testnet" >> $GITHUB_ENV
+          
+      - name: 'Sets env vars for dev-tesnet'
+        if: ${{ github.event.inputs.testnet_type == 'dev-testnet' }}
+        run: |
+          echo "L2_ENCLAVE_DOCKER_BUILD_TAG=testnetobscuronet.azurecr.io/obscuronet/dev_enclave:latest" >> $GITHUB_ENV
+          echo "L2_HOST_DOCKER_BUILD_TAG=testnetobscuronet.azurecr.io/obscuronet/dev_host:latest" >> $GITHUB_ENV
+          echo "L2_CONTRACTDEPLOYER_DOCKER_BUILD_TAG=testnetobscuronet.azurecr.io/obscuronet/dev_contractdeployer:latest" >> $GITHUB_ENV
+          echo "L2_HARDHATDEPLOYER_DOCKER_BUILD_TAG=testnetobscuronet.azurecr.io/obscuronet/dev_hardhatdeployer:latest" >> $GITHUB_ENV
+          echo "RESOURCE_TAG_NAME=devtestnetlatest" >> $GITHUB_ENV
+          echo "RESOURCE_STARTING_NAME=D" >> $GITHUB_ENV
+          echo "RESOURCE_TESTNET_NAME=devtestnet" >> $GITHUB_ENV
+
+      - name: 'Output env vars'
+        id: outputVars
+        run: |
+          echo "L2_ENCLAVE_DOCKER_BUILD_TAG=${{env.L2_ENCLAVE_DOCKER_BUILD_TAG}}" >> $GITHUB_OUTPUT
+          echo "L2_HOST_DOCKER_BUILD_TAG=${{env.L2_HOST_DOCKER_BUILD_TAG}}" >> $GITHUB_OUTPUT
+          echo "L2_CONTRACTDEPLOYER_DOCKER_BUILD_TAG=${{env.L2_CONTRACTDEPLOYER_DOCKER_BUILD_TAG}}" >> $GITHUB_OUTPUT
+          echo "L2_HARDHATDEPLOYER_DOCKER_BUILD_TAG=${{env.L2_HARDHATDEPLOYER_DOCKER_BUILD_TAG}}" >> $GITHUB_OUTPUT
+          echo "RESOURCE_TAG_NAME=${{env.RESOURCE_TAG_NAME}}" >> $GITHUB_OUTPUT
+          echo "RESOURCE_STARTING_NAME=${{env.RESOURCE_STARTING_NAME}}" >> $GITHUB_OUTPUT
+          echo "RESOURCE_TESTNET_NAME=${{env.RESOURCE_TESTNET_NAME}}" >> $GITHUB_OUTPUT
+
+      - name: 'Login to Azure docker registry'
+        uses: azure/docker-login@v1
+        with:
+          login-server: testnetobscuronet.azurecr.io
+          username: ${{ secrets.REGISTRY_USERNAME }}
+          password: ${{ secrets.REGISTRY_PASSWORD }}
+
+      - name: 'Build and push obscuro node images'
+        run: |
+          docker build -t ${{env.L2_ENCLAVE_DOCKER_BUILD_TAG}} -f dockerfiles/enclave.Dockerfile  .
+          docker push ${{env.L2_ENCLAVE_DOCKER_BUILD_TAG}}
+          docker build -t ${{env.L2_ENCLAVE_DOCKER_BUILD_TAG}} -f dockerfiles/host.Dockerfile .
+          docker push ${{env.L2_ENCLAVE_DOCKER_BUILD_TAG}}
```

Comment:
These should be the L2_HOST_DOCKER_BUILD_TAG build tags, not the L2_ENCLAVE_DOCKER_BUILD_TAG one (it's been duplicated)

          docker build -t ${{env.L2_ L2_HOST_DOCKER_BUILD_TAG}} -f dockerfiles/host.Dockerfile .
          docker push ${{env.L2_ L2_HOST_DOCKER_BUILD_TAG}}

Reply:
Fixed !